src/foam/nanos/dig/HTTPDigestSink.java

```diff
@@ -73,6 +73,53 @@ public HTTPDigestSink(String url, String bearerToken, DUGDigestConfig dugDigestC
 
   @Override
   public void put(Object obj, Detachable sub) {
+    try {
+      int responseCode = sendRequest(obj);
+
+      if ( responseCode == HttpServletResponse.SC_OK ) return;
+
+      if ( responseCode == HttpServletResponse.SC_BAD_REQUEST ) { // client error
+        if ( obj instanceof EmailWebhook ) {
+          EmailWebhook ewh = (EmailWebhook) obj ;
+          DAO alarmDAO = (DAO) getX().get("alarmDAO");
+          String emailAddress = ewh.getEmail();
+          String name = "Unable to send request information email to : " + emailAddress;
```

Comment:
do not make alarm name specific, put the name in the notes. The alarm is just to get attention. 
```suggestion
          String name = "EmailWebHook";
```

Reply:
@hchoura please review the suggestion